pkg/pillar/cmd/nim/nim.go

```diff
@@ -782,6 +805,12 @@ func (n *nim) handleDPCImpl(key string, configArg interface{}, fromFile bool) {
 		n.forceLastResort = false
 		n.reevaluateLastResortDPC()
 	}
+	// if device can connect to controller it may get a new DPC in global config. This global DPC
+	// will have higher priority but can be invalid and the device will loose connectivity again
+	// at least temporarily while DPC is being tested. To avoid this we reset the timestamp on
+	// the Manual DPC to the current time
+	// TODO: do it. or check for ManualDPCKey in DPCManager
+	// TODO 2: we should not try lastresort DPC if the user set the DPC to manual
```

Comment:
Is this PR ready for review&merge despite these two TODOs?

Reply:
@milan-zededa I hope yes. It address immediate chicken and egg problem with incorrect global network configuration. I can remove the comment